tests/optimizers/botorch_base_test.py

```diff
@@ -155,3 +156,28 @@ def test_find_optimum_in_1d_discrete_space(seed):
     assert (
         sum(l == 0 for l in losses) > 5
     ), "After figuring out the best of the three points, it should only propose that."
+
+
+def test_get_numerical_points_from_discrete_space():
+    p0l, p0h = -5, 10
+    p1 = ("small", "medium", "large")
+    p2 = ("woof", "miaow", "moo")
+    discrete_space = ps.ParameterSpace()
+    discrete_space.add(ps.IntegerParameter("integ", (p0l, p0h)))
+    discrete_space.add(ps.OrdinalParameter("ordin", p1))
+    discrete_space.add(ps.CategoricalParameter("categ", p2))
+
+    points = _get_numerical_points_from_discrete_space(discrete_space)
+    assert points.shape[0] == 16 * 3 * 3
+    assert points.shape[-1] == 3
+    for integ, ordin, categ in itertools.product(range(p0l, p0h), p1, p2):
+        assert (
+            (
+                points
+                == discrete_space.to_numerical(
+                    dict(integ=integ, ordin=ordin, categ=categ)
+                )
+            )
+            .all(axis=1)
+            .any()
+        )
```

Comment:
I understand the need for a for loop here, but wonder if we can provide a more helpful assertion error message in case this fails. From developing this functionality, do you have an idea what would help or whether the default assertion error is already helpful enough?

Reply:
I provide a suggestion in the [revised version](https://github.com/boschresearch/blackboxopt/pull/86/files/2b325de9a57b60a358d4a7970c2390490c626720..2932ea21b8de502377e0e1ed9b5d379e271a83c8), please let me know your thoughts :)